### DISCUSSION TITLE
Option to select centering of first azimuthal slice in 2d integrator

### DISCUSSION THREAD

**Discussion by djm87** [silx-kit/pyFAI#2338] [Q&A] [answered]
Often there are directions (e.g loading, detector, material, etc.) which are orthogonal and aligned with  0 and 90 degrees along the azimuth. In the analysis (single peak fits, Rietveld, etc.) we want integrated regions that are centered on these special directions and we want the standard 2d integrated results. The 2d integrator automatically centers off of these 0 and 90 directions (e.g. for a 10dg azimuthal bin, the first integration will be centered at 5dg). Playing around with the chi discontinuity and range options for the 2d integrator, it is apparent that you can't make something work without have an azimuthal range greater the 0-2pi or an arbitrary discontinuity. I don't see an option other than using range with a 1d integrator to cobble together a work around.

Am I missing something? 

Thanks, 
Dan

**Comment by kif** [comment, marked as answer]
Hi Dan,
If I understand properly, you would like to integrate around 0°, 90°, 180° and 270° (+/-5°) and due to the 360° periodicity, one of the interesting segment is always cut by the discontinuity.
If this is your problem, you can offset the "zero" position by setting the rot3 to a non null value, for example -45° so that it does not interfere with your region of interest. Of course, your region of interest are now -135, -45, 45, 135.

Can I move this issue to the discussion section ?

**Comment by djm87** [comment]
Hi Kif, 

This is a fine and simple solution.

Thanks for the quick response,
Dan
